numpy/core/src/umath/ufunc_object.c

```diff
@@ -4555,42 +4631,20 @@ ufunc_generic_call(PyUFuncObject *ufunc, PyObject *args, PyObject *kwds)
     /* wrap outputs */
     for (i = 0; i < ufunc->nout; i++) {
         int j = ufunc->nin+i;
-        PyObject *wrap = wraparr[i];
+        _ufunc_context context;
+        PyObject *wrapped;
 
-        if (wrap == NULL) {
-            /* default behavior */
-            retobj[i] = PyArray_Return(mps[j]);
-        }
-        else if (wrap == Py_None) {
-            Py_DECREF(wrap);
-            retobj[i] = (PyObject *)mps[j];
-        }
-        else {
-            PyObject *res;
-            PyObject *args_tup;
+        context.ufunc = ufunc;
```

Comment:
Now I think I've made the comment, but perhaps forgot to submit: maybe just define `_ufunc_context context = {ufunc, full_args, i};`? Or is that incorrect syntax? (My C remains quite hopeless...)

Reply:
Sure, that seems reasonable - Definitely correct in some version of C, I think C90 will cope.